website_docs/automatic_instrumentation.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Automatic Instrumentation
+weight: 3
+---
+
+Automatic instrumentation in Go is done by pulling in dependencies for the component you'd like to have automatically instrumented. For example, you can use the automatic instrumentation package for `net/http` to automatically create spans that track inbound and output requests from your app or service.
+
+## Setup
+
+Each automatic instrumentation package is pulled in as a dependency for your app.
+
+In general, this means you `go get` the appropriate package:
+
+```console
+go get go.opentelemetry.io/contrib/instrumentation/{import-path}/otel{package-name}
+```
+
+And you use it in your code directly (typically wrapping something).
+
+## Example with `net/http`
+
+As an example, here's how you can set up automatic instrumentation for inbound HTTP requests:
+
+First, get the `net/http` automatic instrumentation package:
+
+```console
+go get go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp
+```
+
+Next, wrap an HTTP handler in your code:
+
+```go
+
+// Import the package
+import (
+  // ...
+  "time"
+  "net/http"
+  "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+  // ...
+)
+
+// ...
+
+// Have a function that does some work,
+// which will connect with automatic instruemtnation later in the code
+func sleepy(ctx context.Context) {
+    _, span := tracer.Start(ctx, "sleep")
+    defer span.End()
+
+    sleepTime := 1 * time.Second
+    time.Sleep(sleepTime)
+
+    span.SetAttributes(attribute.Int("sleep.duration", sleepTime))
+}
+
+// Have some http handler function to instrument
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+    fmt.Fprintf(w, "Hello, World! I am instrumented autoamtically!")
+
+    ctx := r.Context()
+
+    // Do some work
+    sleepy(ctx)
+}
+
+// Instantiate and wrap the httpHandler function that you just defined
+handler := http.HandlerFunc(httpHandler)
+wrappedHandler := otelhttp.NewHandler(handler, "hello-instrumented")
+http.Handle("/hello-instrumented", wrappedHandler)
+
+// And start the HTTP server
+log.Fatal(http.ListenAndServe(":3030", nil))
+```
+
+Assuming that you have a `Tracer` and [exporter](exporting_data.md) configured, this code will:
```

Comment:
I don't like that I elided the code for this in the sample, but including it makes the sample considerably larger (creating an exporter, creating a tracer provider, setting up context propagation, etc.). I'm happy to add it, but I feel like linking out to a complete sample is a better choice than embedding it into the docs.